openfl-workspace/torch_cnn_mnist/README.md

```diff
@@ -0,0 +1,162 @@
+* plan: Contains configuration files for federated learning:
+    - plan.yaml: The main Federated Learning plan declaration, defining the structure of the federated learning workflow.
+    - cols.yaml: A list of authorized collaborators for the federated learning task.
+    - data.yaml: Specifies the path to the data set for each collaborator.
+    - defaults: Path to the default configuration values for the federated learning plan.
+* src: This directory contains the Python modules used for federated learning:
+    - init.py: Marks the src directory as a Python package, allowing you to import modules within the directory.
+    - cnn_model.py: Defines the Convolutional Neural Network (CNN) model for federated learning.
+    - dataloader.py: A module responsible for loading and processing datasets for the federated learning task.
+    - taskrunner.py: The core task runner module that manages the execution of federated learning tasks.
+
+## Defining the Data Loader
+The data loader in OpenFL is responsible for batching and iterating through the dataset that will be used for local training and validation on each collaborator node. The TemplateDataLoader class in src/dataloader.py is designed to be a starting template for creating a data loader that is tailored to the FL experiment’s data format requirements.
+
+To customize the TemplateDataLoader, we just need to implement the load_dataset() function to process the dataset available at data_path on the local file system. The data_path parameter comes from the data.yaml configuration file, which is populated when the collaborator’s identity is created via fx collaborator create.
+
+## Defining the Task Runner
+The Task Runner class defines the actual computational tasks of the FL experiment (such as local training and validation). We can implement the placeholders of the TemplateTaskRunner class (src/taskrunner.py) by importing the DigitRecognizerCNN model, as well as the train_epoch() and validate() helper functions from the centralized ML script. The template also provides placeholders for providing custom optimizer and loss function objects.
+
+## How to run this tutorial (local simulation):
+The fx plan initialize command bootstraps the workspace by first setting the initial weights of the aggregate model. It then parses the plan, updates the aggregator address if necessary, and produces a hash of the initialized plan for integrity and auditing purposes.
+
+To help OpenFL calculate the initial model weights, we need to provide the shape of the input tensor as an additional parameter. For the MNIST data set of grayscale (single-channel) 28x28 pixel images, the input tensor shape is [1,28,28]. We will also use a locally deployed aggregator (localhost). Thus, the workspace initialization command for our local federation becomes:
+
+```
+mkdir save
+fx plan initialize --input_shape [1,28,28] --aggregator_address localhost
```

Comment:
This should be Step 1 instead of # Step 1: Setup the Aggregator #.  Also add step to activate virtual environment

Reply:
resolved in d8dda083b7c32d0680ef2d87564c3fa64b888f19